api/client/service/opts.go

```diff
@@ -253,6 +307,21 @@ func (m *MountOpt) Value() []swarm.Mount {
 	return m.values
 }
 
+func getMountType(fields []string) (swarm.MountType, []string) {
```

Comment:
Could the clarify what the mistake is here? Is this the same comment as the above?

The issue with the previous grammar was using a colon to separate fields, when those fields could contain colons, and there was no way to escape it. Also making all the fields positional meant that optional fields were problematic, and adding more fields was even worse.

The only similarity I see here is that the `type` is being required as the first field. This makes parsing significantly easier, and doesn't required any other fields to be positional.